### PR TITLE
Add basic skills - Level 10: Data Generation

### PR DESCRIPTION
**Pull request by bonhunko**
### Summary
Ticket no. 107282
Add basic skills - Level 10: Data Generation

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [x] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: docs/source/docs/level-up/intermediate_skills/10_data_generation.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    With Pthon API, we can generate the synthetic images as below.
+
+    .. code-block:: python
+
+        from datumaro.plugins.synthetic_data import FractalImageGenerator
+
+        FractalImageGenerator(output_dir=<path/to/data>, count=GEN_IMG_COUNT, shape=GEN_IMG_SHAPE).generate_dataset()
+
+    ``GEN_IMG_COUNT`` is an integer that indicates the number of images to be generated. (e.g. `count=300`)
+    ``GEN_IMG_SHAPE`` is a tuple representing the shape of generated images as (width, height) (e.g. `shape=(240, 180))
```
Comment (vinnamkim) [inline docs/source/docs/level-up/intermediate_skills/10_data_generation.rst]:
```suggestion
    ``GEN_IMG_SHAPE`` is a tuple representing the shape of generated images as (width, height) (e.g. ``shape=(240, 180)``)
```

File: docs/source/docs/level-up/intermediate_skills/10_data_generation.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    With Pthon API, we can generate the synthetic images as below.
+
+    .. code-block:: python
+
+        from datumaro.plugins.synthetic_data import FractalImageGenerator
+
+        FractalImageGenerator(output_dir=<path/to/data>, count=GEN_IMG_COUNT, shape=GEN_IMG_SHAPE).generate_dataset()
+
+    ``GEN_IMG_COUNT`` is an integer that indicates the number of images to be generated. (e.g. `count=300`)
```
Comment (vinnamkim) [inline docs/source/docs/level-up/intermediate_skills/10_data_generation.rst]:
```suggestion
    ``GEN_IMG_COUNT`` is an integer that indicates the number of images to be generated. (e.g. ``count=300``)
```

File: docs/source/docs/level-up/intermediate_skills/10_data_generation.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  .. tab-item:: CLI
+
+    We can generate the synthetic images by the following CLI command:
+
+    .. code-block:: bash
+
+      datum generate -o <path/to/data> --count GEN_IMG_COUNT --shape GEN_IMG_SHAPE
+
+    ``GEN_IMG_COUNT`` is an integer that indicates the number of images to be generated. (e.g. `--count 300`)
+    ``GEN_IMG_SHAPE`` is the shape (width height) of generated images (e.g. `--shape 240 180`)
```
Comment (vinnamkim) [inline docs/source/docs/level-up/intermediate_skills/10_data_generation.rst]:
```suggestion
    ``GEN_IMG_COUNT`` is an integer that indicates the number of images to be generated. (e.g. ``--count 300``)
    ``GEN_IMG_SHAPE`` is the shape (width height) of generated images (e.g. ``--shape 240 180``)
```

File: docs/source/docs/level-up/intermediate_skills/10_data_generation.rst [hunk 4]
```diff
@@ -0,0 +1,43 @@
+===========================
+Level 10: Data Generation
+===========================
+
+
+Pre-training of deep learning models for vision tasks can increase model accuracy.
+Training model with the synthetic dataset is one of famouse pre-training approach
+since the manual annotations is quite expensive work.
+
+Base on the `previous research <https://arxiv.org/abs/2103.13023>`_,
```
Comment (vinnamkim) [inline docs/source/docs/level-up/intermediate_skills/10_data_generation.rst]: Please use citation.
https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#citations

**Review by vinnamkim — COMMENTED** [review]
Did you run this CLI command in real to check whether it is working?

**Comment by bonhunko** [comment]
> Did you run this CLI command in real to check whether it is working?

Yes, I manually checked both CLI & Python API

File: docs/source/docs/level-up/intermediate_skills/10_data_generation.rst (hunk 4 above)
Comment (bonhunko) [reply docs/source/docs/level-up/intermediate_skills/10_data_generation.rst]: Cool. I modified to use citation.

**Review by vinnamkim — APPROVED** [review]
LGTM.

**Review by wonjuleee — APPROVED** [review]

**Pull request merged**